lib/browser/browser.ts

```diff
@@ -382,26 +382,46 @@ class OktaAuthBrowser extends OktaAuthBase implements OktaAuth, SignoutAPI {
   // Common Methods from downstream SDKs
   //
 
+  async isAuthenticated(timeout?: number): Promise<boolean> {
+    const authState = this.authStateManager.getAuthState();
+    if (!authState.isPending) {
+      return Promise.resolve(authState.isAuthenticated);
+    }
+
+    let clear, handler, timeoutId;
+    return new Promise(resolve => {
+      clear = () => {
+        this.authStateManager.unsubscribe(handler);
+        clearTimeout(timeoutId);
+      };
+      handler = ({isAuthenticated, isPending}) => {
+        if (!isPending) {
+          resolve(isAuthenticated);
+          clear();
+        }
+      };
+      timeoutId = setTimeout(() => {
+        resolve(false);
```

Comment:
should we clear before resolve?

Reply:
either way should work, resolve won't stop the execution of the code after it.

Reply:
unless it throws?